packages/opentelemetry-sampler/src/InstanaAlwaysOnSampler.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const { SamplingDecision, trace } = require('@opentelemetry/api');
+
+class InstanaAlwaysOnSampler {
+  /**
+   * The Instana always on sampler can only be used in combination with the
+   * Instana propagator.
+   *
+   * The sampler will read the span context and check whether tracing is enabled based
+   * on the incoming Instana headers.
+   */
+  shouldSample(context) {
+    // NOTE: We need to use `RECORD_AND_SAMPLED`, otherwise `traceFlags` will be set to 0
+    let decision = SamplingDecision.RECORD_AND_SAMPLED;
+
+    try {
+      const nonRecordingSpan = trace.getSpan(context).spanContext();
+      if (nonRecordingSpan.traceFlags === 0) {
```

Comment:
*issue(blocking):* This is not the correct way to read the flags from the context. See my change here: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1077/files#diff-11002f6f5094c073f11ebb3418523fa956999dd6637cf0552e907589d9f0f475R75-R77

The W3C trace context spec describes the correct way to read/interprete the flags here: https://www.w3.org/TR/trace-context/#trace-flags:

> As this is a bit field, you cannot interpret flags by decoding the hex value and looking at the resulting number. For example, a flag 00000001 could be encoded as 01 in hex, or 09 in hex if present with the flag 00001000. A common mistake in bit fields is forgetting to mask when interpreting flags.
> 
> Here is an example of properly handling trace flags:
> 
> static final byte FLAG_SAMPLED = 1; // 00000001
> ...
> boolean sampled = (traceFlags & FLAG_SAMPLED) == FLAG_SAMPLED;

Reply:
*remark*: IIRC the plan was to always sample, unless X-INSTANA-L=0 is present. Since we read the sampled flag instead of directly reading the HTTP headers, we would also decide to not sample if any other propagator has created a context and set the sampled flag to 0. AFAICT we cannot do anything else here. So actually the original plan was flawed :-)

The docs say to only use this with the Instana propagator, so we might argue that this is a non-issue.

Anyway, I would leave the behavior here as it is (except for the bit masking change I mentioned above). Maybe we should change the name though? "AlwaysOn" does not really fit what we are doing here, changing this to just InstanaSampler might be better.

Reply:
> The W3C trace context spec describes the correct way to read/interprete the flags here: https://www.w3.org/TR/trace-context/#trace-flags:

Will adapt so it is future proof 👍

Reply:
> should change the name though? "AlwaysOn" does not really fit what we are doing here, changing this to just InstanaSampler might be better.

Not sure 🤔 Maybe we want to add more samplers in the future, then it might be better to give the sampler a more specific name. This sampler is always on except tracing is disabled. But no strong opinion here.

Reply:
I don't have a strong opinion either :-)  So use whatever you think is best. 👍